sros2/test/sros2/verb/test_amend_policy.py

```diff
@@ -0,0 +1,280 @@
+
+
+def test_get_event_permission_for_profile():
+    node_name = NodeName('node', '/ns', '/ns/node')
+    # Get profile
+    policy = etree.fromstring(TEST_POLICY)
+    profile = policy[0]
+
+    assert EventPermission.ALLOW == get_event_permission_for_profile(
+      profile, Event(node_name, 'topic', 'subscribe', 'parameter_events'))
+    assert EventPermission.ALLOW == get_event_permission_for_profile(
+      profile, Event(node_name, 'topic', 'publish', 'parameter_events'))
+    assert EventPermission.ALLOW == get_event_permission_for_profile(
+      profile, Event(node_name, 'service', 'reply', '~get_parameters'))
+    assert EventPermission.NOT_SPECIFIED == get_event_permission_for_profile(
+      profile, Event(node_name, 'topic', 'publish', 'not_a_topic'))
+    assert EventPermission.DENY == get_event_permission_for_profile(
+      profile, Event(node_name, 'topic', 'publish', 'denied_topic'))
+
+
+def test_get_permission_groups():
+    policy = etree.fromstring(TEST_POLICY)
+    profile = policy[0]
+
+    # Permission exists
+    assert len(AmendPolicyVerb.get_permission_groups(
+            profile, 'topic', 'subscribe', 'parameter_events')) == 0
+
+    # Permission doesn't exist
+    assert not AmendPolicyVerb.get_permission_groups(
+            profile, 'topic', 'publish', 'not_a_topic')
+
+
+def test_create_permission():
+    node_name = NodeName('node', '/ns', '/ns/node')
+    # Expression is private to node
+    event = Event(node_name, 'topic', 'publish', '/ns/node/parameter_events')
+    permission_string = b'<topic>~parameter_events</topic>'
+    assert permission_string == etree.tostring(AmendPolicyVerb.create_permission(event))
+
+    # Expression is in node namespace
+    event = Event(node_name, 'topic', 'publish', '/ns/parameter_events')
+    permission_string = b'<topic>parameter_events</topic>'
+    assert permission_string == etree.tostring(AmendPolicyVerb.create_permission(event))
+
+    # Expression is in root namespace with node
+    root_node_name = NodeName('node', '/', '/node')
+    event = Event(root_node_name, 'topic', 'publish', '/parameter_events')
+    permission_string = b'<topic>parameter_events</topic>'
+    assert permission_string == etree.tostring(AmendPolicyVerb.create_permission(event))
+
+    # Expression is in another namespace
+    event = Event(node_name, 'topic', 'publish', '/other_ns/parameter_events')
+    permission_string = b'<topic>/other_ns/parameter_events</topic>'
+    assert permission_string == etree.tostring(AmendPolicyVerb.create_permission(event))
+
+
+def elements_equal(e1, e2):
+    if e1.tag != e2.tag:
+        return False
+    if e1.text != e2.text:
+        return False
+    if e1.tail != e2.tail:
+        return False
+    if e1.attrib != e2.attrib:
+        return False
+    if len(e1) != len(e2):
+        return False
+    return all(elements_equal(c1, c2) for c1, c2 in zip(e1, e2))
+
+
+def test_add_permission():
```

Comment:
Any chance we can migrate these tests to #128?

Reply:
Likely doable.
The whole `test_add_permission` ? And only this one?